lib/linguist/repository.rb

```diff
@@ -96,7 +96,7 @@ def breakdown_by_file
       @file_breakdown ||= begin
         breakdown = Hash.new { |h,k| h[k] = Array.new }
         cache.each do |filename, (language, _)|
-          breakdown[language] << filename
+          breakdown[language] << filename.dup.force_encoding("UTF-8")
```

Comment:
We might want to call `scrub` here.  If the filename contains bytes that aren't valid UTF-8, we'll end up with strings that aren't "valid":

```
irb(main):004:0> path = "/foo/ba\xe9r/baz"
=> "/foo/ba\xE9r/baz"
irb(main):005:0> path.force_encoding "UTF-8"
=> "/foo/ba\xE9r/baz"
irb(main):006:0> path.encoding
=> #<Encoding:UTF-8>
irb(main):007:0> path.valid_encoding?
=> false
irb(main):008:0> path.scrub
=> "/foo/ba�r/baz"
irb(main):009:0> scrubbed = path.scrub
=> "/foo/ba�r/baz"
irb(main):010:0> scrubbed.encoding
=> #<Encoding:UTF-8>
irb(main):011:0> scrubbed.valid_encoding?
=> true
```

Reply:
Oooo. I've added the scrub and also updated the tests to confirm this is working as expected. Thanks @tenderlove.